setup.py

```diff
@@ -169,29 +221,42 @@
         " specify it using the environment variable FMIL_HOME."
     )
 
-if 0 != sys.argv[1].find("clean"): #Dont check if we are cleaning!
-
-    #Check to see if FMILIB_SHARED exists and if so copy it, otherwise raise exception
-    if sys.platform.startswith("win"):
-        dirs_to_search = libdirs + bindirs
-        while dirs_to_search:
-            path_to_dir = os.path.abspath(dirs_to_search.pop())
-            for file_name in os.listdir(path_to_dir):
-                full_path = os.path.join(path_to_dir, file_name)
-                if "fmilib_shared" in file_name and not file_name.endswith(".a"):
-                    fmilib_shared = shutil.copy2(full_path, os.path.join(".", "src", "pyfmi"))
-                    dirs_to_search = None
-                    break
-        if not fmilib_shared:
-            raise Exception(
-                f"Could not find shared library 'fmilib_shared' at either location:" + \
-                    f"\n\t{', '.join(dirs_to_search)}")
-
-        if copy_gcc_lib:
-            path_gcc_lib = ctypes.util.find_library("libgcc_s_dw2-1.dll")
-            if path_gcc_lib is not None:
-                shutil.copy2(path_gcc_lib,os.path.join(".","src","pyfmi"))
-                gcc_lib = os.path.join(".","src","pyfmi","libgcc_s_dw2-1.dll")
+def find_dynamic_fmil_library():
```

Comment:
All for dividing things into smaller functions, but can we then also take e.g., `libdirs` and `bindirs` as explicit arguments?

Reply:
Fixed